l3kernel/l3keys.dtx

```diff
@@ -775,12 +775,14 @@
 % If a key has not previously been defined (is unknown), \cs{keys_set:nn}
 % looks for a special \texttt{unknown} key for the same module, and if this is
 % not defined raises an error indicating that the key name was unknown. This
-% mechanism can be used for example to issue custom error texts.
+% mechanism can be used for example to issue custom error texts. The
+% \texttt{unknown} key also supports the \texttt{.default:n} property.
 % \begin{verbatim}
 %   \keys_define:nn { mymodule }
 %     {
 %       unknown .code:n =
-%         You~tried~to~set~key~'\l_keys_key_str'~to~'#1'.
+%         You~tried~to~set~key~'\l_keys_key_str'~to~'#1'. ,
+%       unknown .default:V = \c_novalue_tl
```

Comment:
Not keen on `\c_novalue_tl` here: that's really an `ltcmd`-helper so I'd rather see something different if we are showing this off.

Reply:
@josephwright I think this is exactly a use case for `\c_novalue_tl`, from the current docs of `\tl_if_novalue:nTF` 

> This function is intended to allow construction of flexible document interface structures in which missing optional arguments are detected.

and

> A marker for the absence of an argument.

`key=value` input is a flexible document interface, and the marker is used for the absence of an argument.

Reply:
@josephwright I think I agree with @Skillmon here. If you think of, say `\foo` being defined with the sig `om` then `\foo{xyz}`would have `#1` being `--NoValue--`. If we extend `\foo`, say, for tagging, and the option argument would become a key value argument with the "default key" for the optional arg being `text` then `\foo{xyz}` would be (or should be) `text= --NoValue--` as would `\foo[bar=abc]{xyz}`, while `\foo[bar=abc,text=123]{xyz}` or \foo[123]{xyz}`would return `text=123`.

So using V of `\c_novalue_tl` makes sense to me in this case.